File: 04-golang/base32.go
/*
----- BASE32.GO -----

To run this program:

  go run base32.go > myfile-0001.txt

Then to use Qrencode:

  qrencode -8 -o qr-myfile-0001.png -r myfile-0001.txt

Or Zint:

  zint -b 58 --eci=26 -o qr-myfile-0001.png -i myfile-0001.txt

*/
package main

import (
  b32 "encoding/base32"
    "fmt"
)

func main() {

// Change the data to suit your needs:
  data := "20231012-1701@kevinleake01"

  uEnc := b32.StdEncoding.EncodeToString([]byte(data))
  fmt.Println(uEnc)
}

File: 03-cpp11/base32.cpp
/*
####################################
#
# --- BASE32.C++ ---
#
####################################

 Example usage:

    gcc base32.cpp; ./a.out > myfile0001.txt

 Then to use Qrencode or Zint:

    qrencode -8 -o qr-myfile-0001.png -r myfile0001.txt
    zint -b 58 --eci=26 -o qr-myfile-0001.png -i myfile0001.txt

*/

#include <cstdlib>

int main(int argc, char *argv[])
{
// Change the data to suit your needs:
  std::system("printf \"20231023-192706@kevinleake01\" | basenc --base32");
}

